iotwifi/commands.go

```diff
@@ -93,3 +93,37 @@ func (c *Command) StartDnsmasq() {
 	cmd := exec.Command("dnsmasq", args...)
 	go c.Runner.ProcessCmd("dnsmasq", cmd)
 }
+
+// StartHostapd starts hostapd.
+func (c *Command) StartHostapd(ssid string, psk string, channel string) {
+	args := []string{
+		"/dev/stdin",
+	}
+	cmd := exec.Command("hostapd", args...)
+
+	cfg := `interface=uap0
+ssid=` + ssid + `
+hw_mode=g
+channel=` + channel + `
+ctrl_interface=/var/run/hostapd
+ctrl_interface_group=0
+macaddr_acl=0
+auth_algs=1
+ignore_broadcast_ssid=0
+wpa=2
+wpa_passphrase=` + psk + `
+wpa_key_mgmt=WPA-PSK
+wpa_pairwise=TKIP
+rsn_pairwise=CCMP`
+
+	c.Log.Info("Hostapd CFG: %s", cfg)
+
+	// handle in pipe here to pass cfg, out/error handled by Runner
+	hostapdPipe, _ := cmd.StdinPipe()
+	hostapdPipe.Write([]byte(cfg))
+
+	go c.Runner.ProcessCmd("hostapd", cmd)
```

Comment:
A possible improvement in the future is to pull the start of these goroutines up into the iotwifi.go file so that all the control logic is in the primary loop. 

I suspect something about how the channels are set up here is what caused the issues we were seeing with logging and communication. Nested threading gets hairy.

Reply:
Makes sense to me.  Saving it for a rainy day.